src/main/java/com/ibm/cloud/cloudant/kafka/mappers/DocumentToSourceRecord.java

```diff
@@ -28,8 +31,12 @@
 
 public class DocumentToSourceRecord implements BiFunction<String, ChangesResultItem, SourceRecord> {
 
-    // Record key schema is Map<String, String>
-    private static final Schema RECORD_KEY_SCHEMA = SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.STRING_SCHEMA).build();
+    // Record key schema is {"_id": doc_id, "cloudant.url": url, "cloudant.db": db}
+    private static final Schema RECORD_KEY_SCHEMA = SchemaBuilder.struct()
+            .field(CloudantConst.CLOUDANT_DOC_ID, Schema.STRING_SCHEMA)
+            .field(InterfaceConst.URL, Schema.STRING_SCHEMA)
+            .field(InterfaceConst.DB, Schema.STRING_SCHEMA)
```

Comment:
Given that ordering is one of the advantages of the struct I wonder if it makes sense to use either:
* url, db, id
or
* id, db, url

that is to say either "least to most" or "most to least" unique, as it might be more intuitive for human use.

Reply:
Interesting. I first did "url, db, id", then thought about a (premature) optimisation where it's probably best to have the most unique part of the key at the beginning, hence the current order.

On that basis, maybe we should do "id, db, url" which is consistent in being "most to least" and (probably) is a bit more optimal when being processed.